quarkus/admin/src/main/java/org/apache/polaris/admintool/config/QuarkusProducers.java

```diff
@@ -76,4 +81,13 @@ public PolarisConfigurationStore configurationStore() {
     // A configuration store is not required when running the admin tool.
     return new PolarisConfigurationStore() {};
   }
+
+  @Produces
+  public List<DataSource> dataSources(@All List<InstanceHandle<DataSource>> dataSources) {
```

Comment:
Is this really needed? I'd expect Quarkus to be able to do this automatically :thinking:

Reply:
This is odd. This method resolves all handles unconditionally. If it is used DataSources have to be configured.

Which test fails without this, specifically?

Reply:
The regression CI fails

Reply:
Please update `JdbcMetaStoreManagerFactory` to require one `DataSource` in the constructor (it only needs one anyway). Then, this producer will not be required.

Let's deal with multiple Data Sources later.

Reply:
yes but if i do that it fails with https://github.com/apache/polaris/pull/1371#discussion_r2047793303, thats why i had to move to Instance<> in Quarkus, so one datasource wouldn't work, never the less we do wanna fix this ASAP and have multiple datasources as an immediate follow-up

Reply:
we do have Pre-Conditions to fail on more than, in the intializeRealm.

Reply:
It worked in my env :) Could you retest?

Reply:
Supporting multiple Data Sources will likely require some changes to the Factory... I do not think it's a simple change from one object to a `List`

Reply:
you mean to say regtest passed in your env ?

Reply:
I tried that approach and it failed again, latest CI for reference : https://github.com/apache/polaris/actions/runs/14675783807/job/41191690984?pr=1371
At this point given we clearly fail when > 1 DS is there and we don;t wanna have Instance logic outside quarkus this parth looks reasonable to me,

Reply:
ah, you mean that docker-based test :thinking:

Reply:
TBH, I only ran the admin tests locally... let's me check the other test.

Reply:
Admin test passes for me as well, its just the regtest fails for me, i think we should be fine given how we gate more than 1 DS and we anways wanna immediately follow-up

Reply:
Let's still go with a single `DataSource` in the factory and limit produces to one `DataSource` too, for now

Reply:
Actually let's use a singular `Instance<DataSource>` in the factory for now and remove DS producers.